sda-commons-server-swagger/src/main/java/org/sdase/commons/optional/server/swagger/parameter/embed/EmbedParameterModifier.java

```diff
@@ -88,9 +98,30 @@ private Model getModelDefinition(Map<String, Model> definitions, String name) {
     return null;
   }
 
+  private Model getResolvedModelDefinition(Map<String, Model> definitions, Model definition) {
+    // In a search endpoint, the resource that supports embedding might be in a list as property.
+    if (!definition.getProperties().containsKey(EMBEDDED_PROPERTY)) {
```

Comment:
Like to see this condition outside of this method to clarify the modelDefinition is retained if it already has embedded properties.